akka-remote/src/main/scala/akka/remote/artery/aeron/ArteryAeronUdpTransport.scala

```diff
@@ -317,45 +319,66 @@ private[remote] class ArteryAeronUdpTransport(_system: ExtendedActorSystem, _pro
         settings.Advanced.Aeron.IdleCpuLevel < 5) 0 // also don't spin for small IdleCpuLevels
     else 50 * settings.Advanced.Aeron.IdleCpuLevel - 240
 
-  override protected def runInboundStreams(): Unit = {
-    runInboundControlStream()
-    runInboundOrdinaryMessagesStream()
+  override protected def runInboundStreams(): Future[Int] = {
+    val bindPort =
```

Comment:
It seems this logic is more complex than before, did you change/fix anything here?

Reply:
it's a little more complicated because we don't have access to the local variable `port` here, and I made it a bit more explicit since that seemed easier to follow.